File: c/sort/bubbleSort.c
#include <stdio.h>
#include <stdbool.h>
#include "util.h"

void bubbleSort(int *arr, int length) {
    for (int i = 0; i < length - 1; i++) {
        bool swapped = false;

        for (int j = 0; j < length - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                int aux = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = aux;
                swapped = true;
            }
        }

        if (!swapped) {
            break;
        }
    }
}

int main() {
    initRandom();

    int arr[10];
    fillArray(arr, 10);

    printf("Antes de ordenar:\n");
    printArray(arr, 10);

    bubbleSort(arr, 10);

    printf("\n\nDepois de ordenar:\n");
    printArray(arr, 10);
    
    printf("\n");

    return 0;
}

File: java/src/main/java/com/marlonzl7/sort/BubbleSort.java
package com.marlonzl7.sort;

import java.util.Random;

public class BubbleSort {
    public static void main(String[] args) {
        int[] array = new int[10];
        Random random = new Random();

        for (int i = 0; i < array.length; i++) {
            array[i] = random.nextInt(100);
        }

        System.out.println("Antes de ordenar:");

        for (int num : array) {
            System.out.print(num + " ");
        }

        bubbleSort(array);

        System.out.println("\nDepois de ordenar:");

        for (int num : array) {
            System.out.print(num + " ");
        }
    }

    public static void bubbleSort(int[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            boolean swapped = false;

            for (int j = 0; j < array.length - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    int aux = array[j];
                    array[j] = array[j + 1];
                    array[j + 1] = array[j];
                    swapped = true;
                }
            }

            if (!swapped) {
                break;
            }
        }
    }
}
